IGaten/ygate.py

```diff
@@ -734,64 +744,70 @@ def start(self):
         """
         signal.signal(signal.SIGINT, self.signal_handler)
         self.start_up()
+        is_ui = re.compile(r" \[.*\] <UI.*>:")
 
         while True:
             if "-i" in str(sys.argv):  # -i as cmd line argument
                 self.aprsis_rx()
-
-            a_p1 = decode_ascii(self.ser.read_until())  # 1st line routing
-            b_p2 = self.ser.read_until()  # 2nd line payload bytes
-            a_p2 = decode_ascii(b_p2)
+            localtime = time.strftime("%H:%M:%S")
+            try:  # in case, serial is disconnected
+                a_p1 = decode_ascii(self.ser.read_until())  # 1st line routing
+                if is_ui.search(a_p1[1]):
+                    b_p2 = self.ser.read_until()  # 2nd line payload bytes
+                else:  # out of sync, disregard payload
+                    b_p2 = b"\r\n"
+                a_p2 = decode_ascii(b_p2)
+            except serial.serialutil.SerialException:
+                print(f"{localtime} {COL.red}Serial read error{COL.end}")
+                logging.error("Serial interface connection error")
+                self.close_pgm()  #exit program
+                break
             routing = a_p1[1]
             payload = a_p2[1]  # non ascii chars will be shown as\xnn
             logging.debug("[FTM ] %s %s", a_p1[1], a_p2[1])
             data_type = self.get_data_type(routing, payload)
             mic_e = mic_e_decode(routing, b_p2)  # mic-e decoding
-            localtime = time.strftime("%H:%M:%S")
 
             if a_p1[0] > 0:  # invalid ascii char in routing
                 print_wrap(
-                    f"{localtime} [INV ] {COL.yellow}Invalid routing: {COL.end} {a_p1[1]}"
+                    f"{localtime} [INV ] "
+                    f"{COL.yellow}Invalid routing: {COL.end} {routing}{payload}"
                 )
                 logging.warning("[INV ] Invalid routing: %s%s", routing, payload)
                 self.pstat[2] += 1
-            elif self.is_routing(routing) and re.search(r" \[.*\] <UI.*>:", routing):
```

Comment:
If this area gets called a lot - there can be some benefit of compiling the regular expression at the beginning - and then use the compiled version.  It CAN be less CPU intensive. Look for re.compile.